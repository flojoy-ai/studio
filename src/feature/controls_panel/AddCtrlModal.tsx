import { Fragment, useState } from "react";
import ReactModal from "react-modal";
import ModalCloseSvg from "../../utils/ModalCloseSvg";
import {
  InputControlsManifest,
  OutputControlsManifest,
} from "./CONTROLS_MANIFEST";
const modalStyles: ReactModal.Styles = {
  overlay: { zIndex: 99 },
  content: {
    border: "1px solid rgba(41, 41, 41, 1)",
    borderRadius: "8px",
    zIndex: 100,
    top: "50%",
    left: "50%",
    transform: "translate(-50%,-50%)",
  },
};
const AddCtrlModal = ({
  isOpen,
  afterOpenModal,
  closeModal,
  addCtrl,
  theme,
}) => {
  const [activeTab, setActiveTab] = useState("input");
  const activeBtnStyle = {
    height: "100%",
    borderBottom:
      theme === "dark"
        ? "2px solid rgb(153, 245, 255)"
        : "2px solid rgba(123, 97, 255, 1)",
  };
  return (
    <ReactModal
      isOpen={isOpen}
      onAfterOpen={afterOpenModal}
      onRequestClose={closeModal}
      style={modalStyles}
      ariaHideApp={false}
      contentLabel="Choose a Ctrl..."
    >
      <button onClick={closeModal} className="close-modal">
        <ModalCloseSvg
          style={{
            height: 23,
            width: 23,
          }}
        />
      </button>
      <div
        className="flex tab-panel"
        style={{
          borderBottom:
            theme === "dark"
              ? "1px solid rgb(47, 46, 46)"
              : "1px solid rgba(217, 217, 217, 1)",
          alignItems: "center",
          gap: "8px",
        }}
      >
        <button
          className="btn-tab"
          onClick={() => setActiveTab("input")}
          style={{
            ...(activeTab === "input" && activeBtnStyle),
          }}
        >
          Inputs
        </button>
        <button
          className="btn-tab"
          onClick={() => setActiveTab("output")}
          style={{
            ...(activeTab === "output" && activeBtnStyle),
          }}
        >
          Outputs
        </button>
      </div>
      {activeTab === "input" && (
        <div
          className="flex"
          style={{
            gap: "16px",
            alignItems: "center",
            width: "fit-content",
            flexWrap: "wrap",
            paddingTop: "12px",
          }}
        >
          {InputControlsManifest.map((ctrl, ctrlIndex) => (
            <Fragment key={ctrlIndex}>
              <button
                className={theme === "dark" ? "cmd-btn-dark" : "cmd-btn"}
                onClick={() =>
                  addCtrl({
                    type: ctrl.type,
                    name: ctrl.name,
                    minWidth: ctrl.minWidth,
                    minHeight: ctrl.minHeight,
                  })
                }
              >
                {ctrl.name}
              </button>
            </Fragment>
          ))}
        </div>
      )}
      {activeTab === "output" && (
        <div
          className="flex"
          style={{
            gap: "16px",
            alignItems: "center",
            width: "fit-content",
            flexWrap: "wrap",
            paddingTop: "12px",
          }}
        >
          {OutputControlsManifest.map((ctrl, ctrlIndex) => (
            <Fragment key={ctrlIndex}>
              <button
                className={theme === "dark" ? "cmd-btn-dark" : "cmd-btn"}
                onClick={() =>
                  addCtrl({
                    type: ctrl.type,
                    name: ctrl.name,
                    minWidth: ctrl.minWidth,
                    minHeight: ctrl.minHeight,
                  })
                }
              >
                {ctrl.name}
              </button>
            </Fragment>
          ))}
        </div>
      )}
    </ReactModal>
  );
};

export default AddCtrlModal;
