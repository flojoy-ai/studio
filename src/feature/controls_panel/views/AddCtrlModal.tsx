export {};
// import { CtlManifestType } from "@src/hooks/useFlowChartState";
// import { Fragment, useState } from "react";
// import ReactModal from "react-modal";
// import ModalCloseSvg from "@src/utils/ModalCloseSvg";
// import {
//   InputControlsManifest,
//   OutputControlsManifest,
// } from "@src/feature/controls_panel/manifest/CONTROLS_MANIFEST";
// import { modalStyles } from "@src/feature/controls_panel/style/ControlModalStyles";
// import { useMantineTheme } from "@mantine/styles";
// import { Box } from "@mantine/core";

// interface AddCtrlModalProps {
//   isOpen: boolean;
//   afterOpenModal: () => void;
//   closeModal: () => void;
//   addCtrl: (ctrlObj: Partial<CtlManifestType>) => void;
// }

// const AddCtrlModal = ({
//   isOpen,
//   afterOpenModal,
//   closeModal,
//   addCtrl,
// }: AddCtrlModalProps) => {
//   const theme = useMantineTheme();
//   const accent =
//     theme.colorScheme === "dark"
//       ? theme.colors.accent1[0]
//       : theme.colors.accent2[0];
//   const [activeTab, setActiveTab] = useState("input");
//   const activeBtnStyle = {
//     height: "100%",
//     borderBottom: `2px solid ${accent}`,
//   };

//   return (
//     <ReactModal
//       isOpen={isOpen}
//       onAfterOpen={afterOpenModal}
//       onRequestClose={closeModal}
//       style={modalStyles(theme)}
//       ariaHideApp={false}
//       contentLabel="Choose a Ctrl..."
//     >
//       <button
//         onClick={closeModal}
//         data-testid={"add-ctrl-modal-close"}
//         className="close-modal"
//       >
//         <ModalCloseSvg
//           style={{
//             height: 23,
//             width: 23,
//           }}
//         />
//       </button>
//       <Box
//         className="flex tab-panel"
//         style={{
//           borderBottom:
//             theme.colorScheme === "dark"
//               ? "1px solid rgb(47, 46, 46)"
//               : "1px solid rgba(217, 217, 217, 1)",
//           alignItems: "center",
//           gap: "8px",
//         }}
//       >
//         <button
//           className="btn-tab"
//           onClick={() => setActiveTab("input")}
//           style={{
//             ...(activeTab === "input" && activeBtnStyle),
//           }}
//         >
//           Inputs
//         </button>
//         <button
//           className="btn-tab"
//           onClick={() => setActiveTab("output")}
//           style={{
//             ...(activeTab === "output" && activeBtnStyle),
//           }}
//         >
//           Outputs
//         </button>
//       </Box>
//       {activeTab === "input" && (
//         <Box
//           className="flex"
//           style={{
//             gap: "16px",
//             alignItems: "center",
//             width: "fit-content",
//             flexWrap: "wrap",
//             paddingTop: "12px",
//           }}
//         >
//           {/* {InputControlsManifest.map((ctrl, ctrlIndex) => (
//             <Fragment key={ctrlIndex}>
//               <button
//                 className={`cmd-btn ${theme}`}
//                 onClick={() =>
//                   addCtrl({
//                     type: ctrl.type,
//                     name: ctrl.name,
//                     minWidth: ctrl.minWidth,
//                     minHeight: ctrl.minHeight,
//                   })
//                 }
//               >
//                 {ctrl.name}
//               </button>
//             </Fragment>
//               )) */}
//         </Box>
//       )}
//       {activeTab === "output" && (
//         <Box
//           className="flex"
//           style={{
//             gap: "16px",
//             alignItems: "center",
//             width: "fit-content",
//             flexWrap: "wrap",
//             paddingTop: "12px",
//           }}
//         >
//           {/* {OutputControlsManifest.map((ctrl, ctrlIndex) => (
//             <Fragment key={ctrlIndex}>
//               <button
//                 className={`cmd-btn ${theme}`}
//                 onClick={() =>
//                   addCtrl({
//                     type: ctrl.type,
//                     name: ctrl.name,
//                     minWidth: ctrl.minWidth,
//                     minHeight: ctrl.minHeight,
//                   })
//                 }
//               >
//                 {ctrl.name}
//               </button>
//             </Fragment>
//           ))}
//         </Box>
//       )}
//     </ReactModal>
//   );
// };

// export default AddCtrlModal;
