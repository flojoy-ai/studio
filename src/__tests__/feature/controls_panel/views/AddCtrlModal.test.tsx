import { fireEvent, render } from "@testing-library/react";
import AddCtrlModal from "@src/feature/controls_panel/views/AddCtrlModal";
import {
  InputControlsManifest,
  OutputControlsManifest,
} from "@src/feature/controls_panel/manifest/CONTROLS_MANIFEST";

describe("AddCtrlModal.tsx", () => {
  const props = {
    isOpen: true,
    afterOpenModal: jest.fn(),
    closeModal: jest.fn(),
    addCtrl: jest.fn(),
  };
  it("render AddCtrlModal correctly.", () => {
    const { container } = render(
      <AddCtrlModal
        isOpen={props.isOpen}
        afterOpenModal={props.afterOpenModal}
        addCtrl={props.addCtrl}
        closeModal={props.closeModal}
        theme={"dark"}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("should fire closeModal function on clicking close button.", () => {
    const { getByTestId } = render(
      <AddCtrlModal
        isOpen={props.isOpen}
        afterOpenModal={props.afterOpenModal}
        addCtrl={props.addCtrl}
        closeModal={props.closeModal}
        theme={"dark"}
      />
    );
    fireEvent.click(getByTestId("add-ctrl-modal-close"));
    expect(props.closeModal).toBeCalledTimes(1);
  });
  it("render all input ctrls in modal by default.", () => {
    const { getByText } = render(
      <AddCtrlModal
        isOpen={props.isOpen}
        afterOpenModal={props.afterOpenModal}
        addCtrl={props.addCtrl}
        closeModal={props.closeModal}
        theme={"dark"}
      />
    );
    InputControlsManifest.forEach((ctrl) => {
      expect(getByText(ctrl.name)).toBeInTheDocument();
    });
  });
  it("should switch to outputs tab on clicking `outputs` button.", () => {
    const { getByText } = render(
      <AddCtrlModal
        isOpen={props.isOpen}
        afterOpenModal={props.afterOpenModal}
        addCtrl={props.addCtrl}
        closeModal={props.closeModal}
        theme={"dark"}
      />
    );
    fireEvent.click(getByText("Outputs"));
    OutputControlsManifest.forEach((ctrl) => {
      expect(getByText(ctrl.name)).toBeInTheDocument();
    });
  });
});
