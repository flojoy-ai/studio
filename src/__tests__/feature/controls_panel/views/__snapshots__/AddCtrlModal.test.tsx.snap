// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCtrlModal.tsx render AddCtrlModal correctly. 1`] = `
<div>
  <div
    data-testid="react-modal"
  >
    <button
      class="close-modal"
      data-testid="add-ctrl-modal-close"
    >
      <svg
        fill="none"
        height="16"
        style="height: 23px; width: 23px;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.1835 11.4905C12.1835 11.6885 12.0845 11.8865 11.8794 12.0915C11.5824 12.3885 10.9814 12.3956 10.6774 12.0915L3.98812 5.2043C3.69114 4.90732 3.68407 4.30628 3.98812 4.00222C4.29218 3.69816 4.88615 3.69816 5.1902 4.00222L11.8794 10.8894C12.0774 11.0874 12.1764 11.2854 12.1835 11.4905Z"
          fill="#BCC2C4"
        />
        <path
          d="M12.1834 4.50427C12.1834 4.70226 12.0844 4.90025 11.9854 5.09824L5.29616 12.0845C4.90018 12.3814 4.39813 12.3885 4.09408 12.0845C3.79002 11.7804 3.79002 11.1864 4.09408 10.8824L10.7833 3.99515C11.0803 3.69816 11.6813 3.69109 11.9854 3.99515C12.0844 4.09414 12.1834 4.29213 12.1834 4.49012L12.1834 4.50427Z"
          fill="#BCC2C4"
        />
      </svg>
    </button>
    <div
      class="flex tab-panel"
      style="border-bottom: 1px solid rgb(47, 46, 46); align-items: center; gap: 8px;"
    >
      <button
        class="btn-tab"
        style="height: 100%; border-bottom: 2px solid rgb(153, 245, 255);"
      >
        Inputs
      </button>
      <button
        class="btn-tab"
      >
        Outputs
      </button>
    </div>
    <div
      class="flex"
      style="gap: 16px; align-items: center; flex-wrap: wrap; padding-top: 12px;"
    >
      <button
        class="cmd-btn dark"
      >
        Text Input
      </button>
      <button
        class="cmd-btn dark"
      >
        Numeric Input
      </button>
      <button
        class="cmd-btn dark"
      >
        Slider
      </button>
      <button
        class="cmd-btn dark"
      >
        Knob
      </button>
      <button
        class="cmd-btn dark"
      >
        Static Numeric Input
      </button>
      <button
        class="cmd-btn dark"
      >
        Dropdown
      </button>
      <button
        class="cmd-btn dark"
      >
        Radio Button Group
      </button>
      <button
        class="cmd-btn dark"
      >
        Checkbox Button Group
      </button>
      <button
        class="cmd-btn dark"
      >
        Local File Loader
      </button>
      <button
        class="cmd-btn dark"
      >
        Node Reference
      </button>
    </div>
  </div>
</div>
`;
